backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Models/SingleRight/CheckDelegationAccess/DelegationAccessCheckResponse.cs

```diff
@@ -0,0 +1,89 @@
+using Newtonsoft.Json;
+
+namespace Altinn.AccessManagement.UI.Core.Models.SingleRight.CheckDelegationAccess
+{
+    /// <summary>
+    ///     Represents the response of the backend for Delegation access check response.
+    /// </summary>
+    public class DelegationAccessCheckResponse
+    {
+        /// <summary>
+        ///     Represents the response of the backend for Delegation access check response.
+        /// </summary>
+        /// <param name="rightKey">The key for the right.</param>
+        /// <param name="resource">The list of resources.</param>
+        /// <param name="action">The action performed.</param>
+        /// <param name="status">The status of the response.</param>
+        /// <param name="faultCode">The fault code, if applicable.</param>
+        /// <param name="reason">The reason for the response.</param>
+        /// <param name="reasonParamsarams">The list of parameters.</param>
+        /// <param name="httpErrorResponse">The optional HTTP error response.</param>
+        public DelegationAccessCheckResponse(
+            string rightKey,
+            List<Resource> resource,
+            string action,
+            string status,
+            string faultCode,
+            string reason,
+            List<ReasonParams> reasonParams,
+            HttpErrorResponse? httpErrorResponse = null)
+        {
+            RightKey = rightKey;
+            Resource = resource;
+            Action = action;
+            Status = status;
+            FaultCode = faultCode;
+            Reason = reason;
+            ReasonParams = reasonParams;
+            HttpErrorResponse = httpErrorResponse;
+        }
+
+        /// <summary>
+        ///     The key for the right.
+        /// </summary>
+        [JsonProperty("rightKey")]
+        public string RightKey { get; init; }
+
+        /// <summary>
+        ///     The list of resources.
+        /// </summary>
+        [JsonProperty("resource")]
+        public List<Resource> Resource { get; init; }
```

Comment:
Not quite sure why it has been made as an object list instead of just an object. But the reason for it being object list here is because it's based on the response object in swagger: https://app.swaggerhub.com/apis/jonkjetiloye/Accessmanagement/1.0.0#/Rights%20delegation/PostRightsDelegationUserDelegationCheck